custom_metrics/almanac.js

```diff
@@ -41,6 +41,36 @@ function parseNodes(nodes) {
   return parsedNodes;
 }
 
+// Return the set of attributes for nodes,
+function getNodesAttributes(nodes) {
+  var object = {};
+  if (nodes) {
+    for (var i = 0, len = nodes.length; i < len; i++) {
+      var node = nodes[i];
+      var attributes = Object.values(getNodeAttributes(node));
+
+      for (var n = 0, len2 = attributes.length; n < len2; n++) {
+        var attribute = attributes[n];
+        object[attribute.name.toLowerCase()] = '';
+      }
+    }
+  }
+  return Object.getOwnPropertyNames(object);
```

Comment:
Seems like this should also be a Set. (please test)
```suggestion
  if (!nodes) {
    return [];
  }
  var uniqueAttributes = new Set();
  for (var node of nodes) {
    var attributes = Object.values(getNodeAttributes(node));
    for (var attribute of attributes) {
      uniqueAttributes.add(attribute.name.toLowerCase());
    }
  }
  return Array.from(uniqueAttributes);
```